src/proto/session.rs

```diff
@@ -172,38 +172,138 @@ impl SessionController {
                     parameters.style, self.options.nickname
                 );
 
-                for (key, value) in parameters.options {
-                    command += format!("{key}={value} ").as_str();
-                }
+                match parameters.style.as_str() {
+                    "STREAM" => {
+                        for (key, value) in parameters.options {
+                            command += format!("{key}={value} ").as_str();
+                        }
+
+                        match &self.options.destination {
+                            DestinationKind::Transient => {
+                                command += "DESTINATION=TRANSIENT ";
+                            }
+                            DestinationKind::Persistent { private_key } => {
+                                command += format!("DESTINATION={private_key} ").as_str();
+                            }
+                        }
+
+                        if !self.options.publish_lease_set {
+                            command += "i2cp.dontPublishLeaseSet=true ";
+                        }
+
+                        command += format!(
+                            "inbound.length={} inbound.quantity={} ",
+                            self.options.inbound_len,
+                            self.options.inbound_qty
+                        )
+                        .as_str();
+
+                        command += format!(
+                            "outbound.length={} outbound.quantity={} ",
+                            self.options.outbound_len,
+                            self.options.outbound_qty
+                        )
+                        .as_str();
+
+                        command += "SIGNATURE_TYPE=7 i2cp.leaseSetEncType=4\n";
+
+                        Ok(command.into_bytes())
+                    }
 
-                match &self.options.destination {
-                    DestinationKind::Transient => {
-                        command += "DESTINATION=TRANSIENT ";
+                    "RAW" => {
+                        for (key, value) in parameters.options {
+                            command += format!("{key}={value} ").as_str();
+                        }
+
+                        command += format!(
+                            "PORT={} HOST={} FROM_PORT={} TO_PORT={} PROTOCOL={} HEADER={}",
+                            self.options.datagram_port,
+                            self.options.datagram_host,
```

Comment:
`PORT` and `HOST` are already set in `parameters.options`, please remove them from this line